### ISSUE TITLE
targetSdkVersion is limited to api 23 at most

### ISSUE THREAD

**Issue by wightwulf1944** [avluis/Hentoid#211]
Currently the app `targetSdkVersion` is set to 23 because it relies on `file:` uris as intent data to open a book in another image viewer such as [Perfect Viewer](https://play.google.com/store/apps/details?id=com.rookiestudio.perfectviewer). Increasing the targetSdkVersion higher than 23 requires that we use `content:` uris instead of `file:` uris and failure to do so will result in a RuntimeException.

When Perfect Viewer is given a `file:` uri referring to an image, Perfect Viewer not only opens the requested image uri but also browses the folder the file is in and allows the user to view all the images in the folder. Even though Perfect Viewer can accept `content:` uris, this only gives permission to the file the uri refers to and not the folder the image file is in. Which means Perfect Viewer cannot view all the images in the directory.

The most common way to get around this is by sending multiple uris and giving access to each file each uri is referring to. This is done by setting the intent action to `ACTION_SEND_MULTIPLE`. However popular image viewers does not seem to accept this intent action.

**Comment by avluis** [comment]
I came across this issue as I was working on the `File` utility classes.
You are correct in your findings.

I also contacted the Perfect Viewer developers;
They mentioned wanting to support `content:` uris natively but no definite date was given.

Most applications that need to support viewing multiple images end up implementing their own image viewers since implementing `ACTION_SEND_MULTIPLE` can get clumsy (permissions, etc).
